utils/build-script-impl

```diff
@@ -641,7 +641,10 @@ if [[ "$SKIP_TEST_WATCHOS" ]] ; then
     SKIP_TEST_WATCHOS_SIMULATOR=1
 fi
 
-if [[ "${CMAKE_GENERATOR}" == "Ninja" ]] && [[ -z "$(which ninja)" ]] ; then
+# Some Linux distributions, e.g. Fedora, use 'ninja-build' instead of
+# 'ninja' as binary name. Check for both.
+if [[ "${CMAKE_GENERATOR}" == "Ninja" ]] && [[ -z "$(which ninja)" ]] &&
+   [[ -z "$(which ninja-build)" ]] ; then
```

Comment:
I'm not terribly familiar with bash, in fact I know enough of it to be dangerous but I have only ninja-build (and not ninja) and the test works as expected on my machine.

$ which ninja
/usr/bin/which: no ninja in (/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/sbin:/home/davide/.local/bin:/home/davide/bin)
[davide@Host-001 ~]$ which ninja-build
/usr/bin/ninja-build